docs/book/how-to/manage-the-zenml-server/best-practices-upgrading-zenml.md

```diff
@@ -0,0 +1,54 @@
+# Best practices for upgrading ZenML
+
+While upgrading ZenML is generally a smooth process, there are some best practices that you should follow to ensure a successful upgrade. Based on experiences shared by ZenML users, here are some key strategies and considerations.
+
+## Upgrading your server
+
+Follow the tips below while upgrading your server to mitigate data losses, downtime and other issues.
+
+### Data Backups
+
+- **Database Backup**: Before upgrading, create a backup of your MySQL database. This allows you to rollback if necessary.
+- **Automated Backups**: Consider setting up automatic daily backups of your database for added security. Most managed services like AWS RDS, Google Cloud SQL, and Azure Database for MySQL offer automated backup options.
+
+![Screenshot of backups in AWS RDS](../../.gitbook/assets/aws-rds-backups.png)
+
+### Upgrade Strategies
+
+- **Staged Upgrade**: For large organizations or critical systems, consider using two ZenML server instances (old and new) and migrating services one by one to the new version.
+    ![Server Migration Step 1](../../.gitbook/assets/server_migration_1.png)
+
+    ![Server Migration Step 2](../../.gitbook/assets/server_migration_2.png)
+
+- **Team Coordination**: If multiple teams share a ZenML server instance, coordinate the upgrade timing to minimize disruption.
+- **Separate Tenants**: Coordination between teams might be difficult if one team requires new features but the other can't upgrade yet. In such cases, it is recommended to use dedicated ZenML server instances per team or product to allow for more flexible upgrade schedules.
+
+### Minimizing Downtime
+
+- **Upgrade Timing**: Plan upgrades during low-activity periods to minimize disruption.
+
+- **Avoid Mid-Pipeline Upgrades**: Be cautious of automated upgrades or redeployments that might interrupt long-running pipelines.
+
+## Upgrading your code
+
+Sometimes, you might have to upgrade your code to work with a new version of ZenML. This is true especially when you are moving from a really old version to a new major version. The following tips might help, in addition to everything you've learnt in this document so far.
+
+### Testing and Compatibility
+
+- **Local Testing**: It's a good idea to update your local server first and run some old pipelines to check for compatibility issues between the old and new versions.
+- **End-to-End Testing**: You can also develop simple end-to-end tests to ensure that the new version works with your pipeline code and your stack. ZenML already has an [extensive test suite](https://github.com/zenml-io/zenml/tree/main/tests) that we use for releases and you can use it as an example.
+- **Artifact Compatibility**: Be cautious with pickle-based [materializers](../../how-to/handle-data-artifacts/handle-custom-data-types.md), as they can be sensitive to changes in Python versions or libraries. Consider using version-agnostic materialization methods for critical artifacts.
```

Comment:
Maybe ask people to try and load older artifacts after upgrading to ensure things go smoothly?

Reply:
added that and a code block